_posts/2019/2019-10-10-building-a-drawer-component-in-react.md

```diff
@@ -0,0 +1,805 @@
+
+## Expanded state and styling
+
+Remember our `is-expanded` class above? We're technically still adding/removing
+it based on the state of `active` but right now it doesn't do anything. It's still nice to keep the functionality as it let's the front-end know the state of the component for styling purposes. We can use this to
+give our Drawer component a bit more flash by providing a simple triangular
+indicator to signal if the Drawer is expanded or collapsed.
+
+First we'll add a new element to our `render()` function that will contain our arrow indicator. We'll
+also surround the displayed `title` inside the button with a `<span>` tag. This will come in handy
+for alignment in a minute.
+
+``` jsx
+<button
+  className="Drawer-title"
+  onClick={this.handleTitleClick}
+>
+  <span className="Drawer-titleText">{title}</span>
+  <div className="Drawer-arrow Drawer--rightArrow" />
+</button>
+```
+
+Now that we've added an element to style let's make it look like an arrow. You
+could use SVG here but we can just as easily do this in CSS. [Chris Coyier of CSS-Tricks has
+an excellent article on how this works](https://css-tricks.com/snippets/css/css-triangle/){:target="_blank"}. I highly recommend a read through.
+
+Here's the CSS we'll need to achieve the effect:
+
+``` scss
+.Drawer {
+  $arrowSize: 5px;
+
+  &-arrow {
+    border-top: $arrowSize solid transparent;
+    border-bottom: $arrowSize solid transparent;
+    border-left: $arrowSize solid #454545;
+    width: 0;
+    height: 0;
+  }
+}
+```
+
+A quick explanation of the above:
+
+* We set the element to have `height` and `width` of 0. This is because adding a border
+to a 0 sized element makes the borders meet at the center of the element like a triangle.
+* `border-top` and `border-bottom` are set to transparent to hide the top and bottom of the triangle
+* `border-left` adds a border on the left side but because it is triangular it appears to be
+pointing to the right.
+
+![Drawer arrow broken](/img/2019/drawer-arrow-off.png)
+
+We've now got a triangle indicator but it isn't positioned properly as seen above. Let's move this to
+the right side of the button using Flexbox.
+
+* First we'll give the parent element `.Drawer-title` the property of `display: flex`.
+* Then we'll set the actual title text element above to flex as much as possible using `flex: 1`.
+* Finally, a small tweak to our arrow by using `align-self: center` make it appear on the same baseline as the bottom of our text characters. Here's the resulting CSS:
+
+
+``` scss
+.Drawer {
+
+  &-title {
+    // omitted
+
+    display: flex;
+  }
+
+  &-titleText {
+    flex: 1;
+  }
+
+  &-arrow {
+    border-top: $arrowSize solid transparent;
+    border-bottom: $arrowSize solid transparent;
+    border-left: $arrowSize solid #454545;
+    width: 0;
+    height: 0;
+
+    align-self: center;
+  }
+}
+```
+
+We can also go one more step and animate this alongside the content animation. This
```

Comment:
We can take one more step and animate this alongside the content animation. This